src/main/java/com/exasol/adapter/request/AdapterRequest.java

```diff
@@ -33,4 +36,14 @@
      * @return Virtual Schema name
      */
     public String getVirtualSchemaName();
+
+    /**
+     * Execute an adapter call.
+     * 
+     * @param adapterCallExecutor instance of {@link AdapterCallExecutor}
+     * @param metadata            metadata for the context in which the adapter exists
+     * @return response in a JSON format
+     * @throws AdapterException if some problem occurs
+     */
+    public String execute(AdapterCallExecutor adapterCallExecutor, ExaMetadata metadata) throws AdapterException;
```

Comment:
I would rather call this method `executeWith` to explicit that the execution is deferred to the `adapterCallExecutor` passed.